app/actions/oauth.go

```diff
@@ -47,6 +48,18 @@ func (action *CreateEditOAuthConfig) IsAuthorized(ctx context.Context, user *ent
 func (action *CreateEditOAuthConfig) Validate(ctx context.Context, user *entity.User) *validate.Result {
 	result := validate.Success()
 
+	if action.Status == enum.OAuthConfigDisabled {
+		tenant := ctx.Value(app.TenantCtxKey).(*entity.Tenant)
+		activeProviders := &query.ListActiveOAuthProviders{}
+		if err := bus.Dispatch(ctx, activeProviders); err != nil {
+			return validate.Failed("Cannot retrieve OAuth providers")
+		}
+
+		if !tenant.IsAllowingEmailAuth && len(activeProviders.Result) == 1 {
+			result.AddFieldFailure("status", "You cannot disable this provider with neither email auth nor any other provider enabled.")
```

Comment:
is there a situation where the UI would allow this to happen?

Reply:
In theory, no. But users may force the checkbox, or maybe a bad synchronization due to a poor connection from the client would cause him to disable email authentication switch and then rush into the OAuth provider edit form (I don't know if it's really possible but I think so, with a very slow connection I think), so I prefer to add this check as we never know what could possibly happen.
Or even worse: Another admin disable the email authentication while the other is on the OAuth provider edit form, and then his UI become out of sync with the server.